### ISSUE TITLE
Validation annotations are not added for primitive definitions referenced in other objects

### ISSUE THREAD

**Issue by srikanthpatlolla** [swagger-api/swagger-codegen#8044]
##### Description

We are trying to re-use some variable declarations which are used across multiple operations with in a single API with json schema restrictions for validation annotations.

When we reference any string or number variable definitions in othe object types in swagger, the validation annotations are not added in java POJO calsses.

##### Swagger-codegen version
2.2.3

##### Swagger declaration file content

```yaml
definitions:
  stringPattern:
    type: string
    minLength: 1
    maxLength: 20
    pattern: '^([a-zA-Z0-9])*$'
  decimalNumber:
    type: number
    minimum: 0
    maximum: 9999999999999.99
    multipleOf: 0.01
  ApiRequest:
    type: object
    properties:
      stringPattern:
        $ref: '#/definitions/stringPattern'
      decimalNumber:
        $ref: '#/definitions/decimalNumber'
      inlineString:
        type: string
        minLength: 1
        maxLength: 20
        pattern: '^([a-zA-Z0-9])*$'
      inlineDecimal:
        type: number
        minimum: 0
        maximum: 9999999999999.99
        multipleOf: 0.01
```

##### Command line used to generate
Running the maven generate-sources with below config options in plugin.
language:java
configOptions
  useBeanValidation:true
  recursiveBeanValidation:true

<b>Generated Java class </b>:
   /**
   * Get stringPattern
   * @return stringPattern
  **/
  @ApiModelProperty(value = "")
  public String getStringPattern() {
    return stringPattern;
  }
   /**
   * Get decimalNumber
   * @return decimalNumber
  **/
  @Valid
  @ApiModelProperty(value = "")
  public BigDecimal getDecimalNumber() {
    return decimalNumber;
  }

   /**
   * Get inlineString
   * @return inlineString
  **/
 @Pattern(regexp="^([a-zA-Z0-9])*$") @Size(min=1,max=20)  @ApiModelProperty(value = "")
  public String getInlineString() {
    return inlineString;
  }
   /**
   * Get inlineDecimal
   * minimum: 0
   * maximum: 9999999999999.99
   * @return inlineDecimal
  **/
  @Valid
 @DecimalMin("0") @DecimalMax("9999999999999.99")  @ApiModelProperty(value = "")
  public BigDecimal getInlineDecimal() {
    return inlineDecimal;
  }

**Comment by FatCash** [comment]
Hi @srikanthpatlolla, I am struggling with the same issue: https://github.com/swagger-api/swagger-codegen/issues/8001 and https://github.com/swagger-api/swagger-codegen/issues/7980
Have not received any replies from devs yet, please share if you figure out a way to fix this!

**Comment by srikanthpatlolla** [comment]
@FatCash : My apologies for the delay, I was out on vacation. I had to duplicate the element definition everywhere. It's error prone and more work, but it was a need for me to complete my project in time,

**Comment by FatCash** [comment]
Kind devs fixed this problem in swagger-codegen-cli-2.4.x

**Comment by srikanthpatlolla** [comment]
Closing this as it is resolved in 3.1.0-SNAPSHOT branhc
